exp/runtime/topologymutation/variables.go

```diff
@@ -53,3 +56,46 @@ func GetStringVariable(templateVariables map[string]apiextensionsv1.JSON, variab
 	}
 	return stringValue, true, nil
 }
+
+// GetBoolVariable get the value as a bool.
+func GetBoolVariable(templateVariables map[string]apiextensionsv1.JSON, variableName string) (bool, bool, error) {
+	value, found, err := GetVariable(templateVariables, variableName)
+	if !found || err != nil {
+		return false, found, err
+	}
+
+	// Unquote the JSON string.
+	stringValue, err := strconv.Unquote(string(value.Raw))
+	if err != nil {
+		return false, true, err
+	}
+	// Parse the JSON bool.
+	boolValue, err := strconv.ParseBool(stringValue)
+	if err != nil {
+		return false, true, err
+	}
+	return boolValue, true, nil
+}
+
+// GetVariableObject returns get variable's string value then unmarshal it into
+// object passed from v.
+func GetVariableObject(templateVariables map[string]apiextensionsv1.JSON, variableName string, v interface{}) (bool, error) {
```

Comment:
I think we should to change the name of this function to differentiate it from the `Bool` and `String` functions which return the variable. 

Maybe `GetObjectVariableInto`? and change the `v` to `into`